### ISSUE TITLE
Native Stack Navigator screen transitions don't match system transitions on Android 13

### ISSUE THREAD

**Issue by philiplindberg** [software-mansion/react-native-screens#1640]
### Description

When using the [Native Stack Navigator](https://reactnavigation.org/docs/native-stack-navigator) from React Navigation, the navigator should use the native APIs (`UINavigationController` on iOS and `Fragment` on Android) to match the system screen transitions for each platform. On iOS this works as expected, but on Android (specifically Android 13), the Native Stack Navigator screen transitions differ significantly from the system transitions as seen in the screen recordings below (recorded on a Google Pixel 6):

https://user-images.githubusercontent.com/1865904/202173547-b0e6960f-4e78-44f6-a395-138dfb2e6ae1.mp4

https://user-images.githubusercontent.com/1865904/202174468-8486f0d3-40d0-4f95-8657-64bc99e13689.mp4

I've also created a Snack that highlights this problem:
https://snack.expo.dev/@philiplindberg/react-native-screens-repro?platform=android

### Steps to reproduce

1. Run provided Snack on an emulator or physical device running Android 13
2. Notice how screen transitions differ significantly from the system transitions (like in the Settings app)

### Snack or a link to a repository

https://snack.expo.dev/@philiplindberg/react-native-screens-repro?platform=android

### Screens version

3.18.2

### React Native version

0.70.5

### Platforms

Android

### JavaScript runtime

JSC

### Workflow

Expo managed workflow

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

Real device

### Device model

Google Pixel 6

### Acknowledgements

Yes

**Comment by hirbod** [comment]
This is kinda expected, since RNS uses these https://github.com/software-mansion/react-native-screens/tree/main/android/src/main/res/anim for animating.

**Comment by kkafar** [comment]
Hi @philiplindberg,

thank you for reporting!

There was a time we used system transitions on Android, but due to changes introduced in React Native 0.66.0 we had to go with custom animation definitions (which mimicked system ones) (see https://github.com/software-mansion/react-native-screens/pull/1213 for explanation why this change had to happen). 
"Unfortunately" Android 13 changed animations -> we're out of sync now and only sensible workaround *for now* is the one mentioned by @hirbod [in the post above](https://github.com/software-mansion/react-native-screens/issues/1640#issuecomment-1319832905).

I'll look into possibility of adding check on API level and using system animations for newer API versions - if there aren't any problems with `React Native` any more. If case there are, I 'll try do add new custom animations which mimic these from new Android versions and use them based on Android API level.
